### ISSUE TITLE
[Bug] [Task Definition] cannot change  Environment Name when create or edit task

### ISSUE THREAD

**Issue by arstercz** [apache/dolphinscheduler#12457]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

It's not work when I create or change the task `Environment Name`, and `Environment Name` always use the first Environment of the Worker Group.

### What you expected to happen

`Environment Name` should be changed so that I can use different `Environment Name` from the same `Worker group`.

### How to reproduce

It's simple to reproduce this issue

    1. into the `Security -> Environment Manage`, create multiple `Environment Name` that all belongs to one worker group. 
    2. into the `Project -> Task -> Task Definition`, change the `Environment Name` by create or edit task;
    3. save and review the task，and will find it's not work;

### How often does this problem occur?

 > every time

### The following instructions may be useful

Our dolphincheduler instance use MySQL database, and found the sql update with wrong contition:
```
# 1
select 
  id, code...
  from t_ds_task_definition
  WHERE code = ...
  and version = ..

# 2
insert into t_ds_task_definition_log ...

# 3
  UPDATE t_ds_task_definition  SET code=...
...
WHERE id=?
```

we cann't find the `id value` from the table `t_ds_task_definition`, but found in table `t_ds_task_definition_log`.

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by hstdream** [comment]
According to your reproduction steps, this problem is not reproduced.
before:
<img width="225" alt="1666250046178" src="https://user-images.githubusercontent.com/33045461/196881162-4ed42a70-7eb5-4dfe-9f8d-367d9195f5ac.png">
after:
<img width="220" alt="1666250118306" src="https://user-images.githubusercontent.com/33045461/196881346-e19e285d-537d-4aba-b7c1-888630a03b2d.png">

**Comment by arstercz** [comment]
> 

We use the standalone mode(v3.1.0).  and found `t_ds_task_definition ` update correctly, but display in the web always wrong.

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/1954680/196921066-c0748f05-bb3a-452e-bb98-7c3bdb1656eb.png">

![image](https://user-images.githubusercontent.com/1954680/196921696-4be0b737-3ab5-4377-bc7e-9478b3ca7e7e.png)

**Comment by hstdream** [comment]
I use the standalone mode(v3.1.0) too.

**Comment by hdygxsj** [comment]
I can't reproduce too ,@arstercz can you provide more infomation to help reproducing this issue ?

**Comment by mer2** [comment]
which milestone will it be fixed? it seems it's a UI bug, when UI dialog show, it should display the corrent env-name last edited. 3q

**Comment by mer2** [comment]
这个BUG随手就可以改掉的，为何要一推再推呢？！

**Comment by hdygxsj** [comment]
> 这个BUG随手就可以改掉的，为何要一推再推呢？！

This issue will be fixed after pr #13148 is completed
